### PR TITLE
This commit further refactors existing work to make windows LDAP and …

### PR DESCRIPTION
**Pull request by alistanis**
…Certificate services available to other parts of Teleport. In particular, this PR moves the majority of the LDAP and CA services from windows_server.go into the windows package and into a new structure CertificateAuthority

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: lib/auth/windows/ldap.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+		if sb.Len() > 0 {
+			sb.WriteString(",")
+		}
+		sb.WriteString("DC=")
+		sb.WriteString(p)
+	}
+	return sb.String()
+}
+
+// ComputerAttributes are the attributes we fetch when discovering
```
[the remark this answers is not in the document]
Comment (alistanis) [reply lib/auth/windows/ldap.go:75]: I see your point there. That makes sense.

File: lib/auth/windows/certificate_authority.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+)
+
+// NewCertificateAuthority returns a new structure for modifying windows certificates in a windows CA
+func NewCertificateAuthority(clusterName string,
+	ldapConfig LDAPConfig,
+	ldapClient *LDAPClient,
+	ap auth.WindowsDesktopAccessPoint,
+	log logrus.FieldLogger) *CertificateAuthority {
```
Comment (smallinsky) [inline lib/auth/windows/certificate_authority.go:34]: nit: Why not just: 
```suggestion
func NewCertificateAuthority(cfg CertificateAuthorityConfig) (*CertificateAuthority, error) {
 if err := cfg.ValidateAndSetDefault(); err != nil {
     return nil, err
 }
 return CertificateAuthority{cfg: cfg}, nil
}
```
where `clusterName` and `lc` can be part of `CertificateAuthorityConfig`

File: lib/srv/desktop/windows_server.go [hunk 3]
```diff
@@ -351,6 +364,8 @@ func NewWindowsService(cfg WindowsServiceConfig) (*WindowsService, error) {
 		s.cfg.Log.Infoln("desktop discovery via LDAP is disabled, set 'base_dn' to enable")
 	}
 
+	s.ca = windows.NewCertificateAuthority(s.clusterName, s.cfg.LDAPConfig, s.lc, s.cfg.AccessPoint, s.cfg.Log)
```
Comment (smallinsky) [inline lib/srv/desktop/windows_server.go:367]: nit: What would you cay about renaming `NewCertificateAuthority` to something like `NewStoreClient` ? 

At first glance `ca` and  `NewCertificateAuthority` naming is a bit misleading for a reader and indicatate the ca contains some kind of `CertificateAuthority` object but actually the ca is a windows store client that allows to update ca in that store.

**Review by smallinsky — APPROVED** [review]

File: lib/auth/windows/certificate_authority.go (hunk 2 above)
Comment (alistanis) [reply lib/auth/windows/certificate_authority.go:34]: Sounds very reasonable.

File: lib/srv/desktop/windows_server.go (hunk 3 above)
Comment (alistanis) [reply lib/srv/desktop/windows_server.go]: I think that's a good idea.

**Pull request merged**